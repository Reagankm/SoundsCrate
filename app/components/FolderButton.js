/* @flow */
import React, { Component } from 'react';
import { withNavigation } from 'react-navigation';
import {
  TouchableOpacity,
  Image,
  View,
  Text,
  StyleSheet,
} from 'react-native';
import type { Folder } from '../discogsTypes';

type Props = {
  folder: Folder,
  thumbnails: Array<string>
};

class FolderButton extends Component {
  props: Props;

  render(): React.Element<*> {
    const { folder, thumbnails } = this.props;
    return (
      <View style={styles.folderContainer}>
        <View style={styles.thumbsContainer}>
          {thumbnails.map((thumb, index) => (
            <Image
              key={index}
              style={styles.folderThumb}
              source={{ url: thumb }} />
          ))}
        </View>
        <TouchableOpacity
          testID={'folderButton'}
          style={styles.folderTouchable}
          onPress={() => this.openFolder(folder.id)}>
          <View style={styles.folderTextWrapper}>
            <Text style={styles.folderText}>{folder.name}</Text>
            <View style={styles.textCountWrapper}>
              <Text style={styles.folderTextCount}>{folder.count}  </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }

  openFolder(id: string) {
    const { navigate } = this.props.navigation;
    navigate('Folder', { id });
  }
}

export default withNavigation(FolderButton);

const styles = StyleSheet.create({
  folderContainer: {
    flex: 1,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    height: 100,
  },
  thumbsContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
    opacity: 0.7,
  },
  folderThumb: {
    flex: 1,
    width: 50,
    height: 100,
  },
  folderTouchable: {
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    zIndex: 1,
  },
  folderTextWrapper: {
    flexDirection: 'row',
  },
  folderText: {
    fontSize: 25,
    fontWeight: 'bold',
    padding: 5,
  },
  textCountWrapper: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  folderTextCount: {
    paddingVertical: 12,
    fontSize: 15,
  },
});
