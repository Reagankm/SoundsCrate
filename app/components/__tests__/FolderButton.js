/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import FolderButton from '../FolderButton';

jest.mock('react-navigation', () => ({
  __esModule: true,
  withNavigation: cls => cls
}));
jest.mock('TouchableOpacity', () => 'TouchableOpacity');

describe('app/components/FolderButton', () => {
  it('should render', () => {
    const tree = renderer.create(
      <FolderButton
        thumbnails={[ 'foo', 'bar' ]}
        folder={{ id: 1, name: 'foo', count: 2 }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should navigate to another page onPress', () => {
    const navigate = jest.fn();
    const tree = renderer.create(
      <FolderButton
        navigation={{ navigate }}
        thumbnails={[ 'foo', 'bar' ]}
        folder={{ id: 1, name: 'foo', count: 2 }}
      />
    ).toJSON();
    // Replace with tree.findByProps({ testID: 'folderButton' }) after upgrading
    // react-test-renderer
    tree.children[1].props.onPress();
    expect(navigate).toHaveBeenCalledWith('Folder', { id: 1 });
  });
});
