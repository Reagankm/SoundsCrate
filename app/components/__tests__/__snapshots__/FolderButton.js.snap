exports[`app/components/FolderButton should render 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#ccc",
      "borderBottomWidth": 1,
      "flex": 1,
      "height": 100,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "opacity": 0.7,
      }
    }>
    <Image
      source={
        Object {
          "url": "foo",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 100,
          "width": 50,
        }
      } />
    <Image
      source={
        Object {
          "url": "bar",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 100,
          "width": 50,
        }
      } />
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 1,
      }
    }
    testID="folderButton">
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "padding": 5,
          }
        }>
        foo
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "paddingVertical": 12,
            }
          }>
          2
            
        </Text>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
